emptyStrCmp/emptyStrCmp.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package emptystrcmp
+
+import (
+	"go/ast"
+	"go/token"
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "emptyStrCmp",
+	Doc:  "check for len(s) == 0 and len(s) != 0 where s is string",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch node.(type) {
+			case *ast.BinaryExpr:
+				b := node.(*ast.BinaryExpr)
+				if callExpr, ok := b.X.(*ast.CallExpr); ok {
+					if idt, ok := callExpr.Fun.(*ast.Ident); ok && idt.Name == "len" &&
+						(b.Op == token.EQL || b.Op == token.NEQ) {
+						arg0 := callExpr.Args[0]
```

Comment:
Check length of `Args` for robustness.

Reply:
I think calling len without any argument will be compilation error, won't it? And len only accepts one argument.

Reply:
Perhaps there is some confusion. I am suggesting to check `len(callExpr.Args) > 0` before accessing the first element. Would that give a compilation error?

Reply:
No what I meant is that args[0] code will be executed only when callexpr is len and len always has one argument so no need to check for length of args slice. If some program is using len without providing a single argument then ilthat program won't compile only.

Reply:
Ah I see. That's why I said for "robustness". You are relying on the Go AST parser to be correct. What if there is a parsing bug due to which the `Args` slice is not properly populated? Obviously the chances are very slim, but it's a good practice to add these safety checks.